lib/client.py

```diff
@@ -166,7 +210,9 @@ def authenticate(self, session, base_url, username, password):
             data_path="CyberArkLogonResult",
             data=auth_post_data,
         )
-        return self._authorization
+        print(self._authorization)
+        yield self._authorization
```

Comment:
the yield part needs to be wrapped in a try-finally block, (with self.logoff in the finally part) to ensure that self.logoff will be called.